lib/pf/web/device_registration.pm

```diff
@@ -18,65 +18,95 @@ use warnings;
 use pf::Authentication::constants;
 use List::MoreUtils qw(any);
 
-Readonly our @DEVICE_OUI => _load_file_into_array($allowed_device_oui_file);
-Readonly our @DEVICE_TYPES => _load_file_into_array($allowed_device_types_file);
-
 =head1 SUBROUTINES
 
-=over
+=cut
+
+=item mac_vendor_id
+
+Get the matching mac_vendor_id from Fingerbank
 
 =cut
 
+sub mac_vendor_id {
+    my ($mac) = @_; 
+    my $logger = get_logger();
+
+    my ($status, $result) = fingerbank::Model::MAC_Vendor->find([{ mac => $mac }, {columns => ['id']}]);
+
+    if(is_success($status)){
+        return $result->id;
+    }else {
+        $logger->debug("Cannot find mac vendor ".$mac." in the database");
+    }
+}
 
-=item _load_file_into_array
+=item device_id
 
-Loads each line of file into array
-Trimming spaces and removing shell style comments from each line
+Get the matching device_id from Fingerbank
 
 =cut
 
-sub _load_file_into_array {
-    my ($file_name) = @_;
-    my @items;
-    if(-r $file_name) {
-        local *FILE;
-        open(FILE,$file_name);
-        @items =
-            grep {$_}
-            map {
-                #Getting rid of newlines, comments and trimming spaces
-                chomp; s/#.*$//; s/^\s+//; s/\s+$//;
-                $_;
-            } <FILE>;
-        close(FILE);
+sub device_id {
+    my ($mac_vendor_id) = @_; 
+    my $logger = get_logger();
+
+    my ($status, $result) = fingerbank::Model::Combination->find([{ mac_vendor_id => $mac_vendor_id }, {columns => ['device_id']}]);
+
+    if(is_success($status)){
+        return $result->device_id;
+    }else {
+        $logger->debug("Cannot find matching device id ".$result->device_id." for this mac vendor id ".$mac_vendor_id." in the database");
     }
-    return @items;
 }
 
-=item is_allowed
+=item is_allowed 
 
-Validate if mac address is an allowed device mac
+Verify 
 
-=cut
+=cut 
 
 sub is_allowed {
     my ($mac) = @_;
-    return 1 unless @DEVICE_OUI;
-    $mac =~ s/O/0/i;
-    $mac = clean_mac($mac);
-    return any { $mac =~ /^\Q$_\E/i } @DEVICE_OUI;
+    my $logger = get_logger();
+    my @oses = @{$Config{'device_registration'}{'oses'}};
+
+    #if no oses are defined then it will not match any oses
+    return $FALSE if @oses == 0;
+
+    $mac =~ s/://g;
+    my $mac_vendor = substr($mac, 0,6);
+    my $mac_vendor_id = mac_vendor_id($mac_vendor);
+    my $device_id = device_id($mac_vendor_id);
+    my ($status, $result) = fingerbank::Model::Device->find([{ id => $device_id}, {columns => ['name']}]);
+
+    # We are loading the fingerbank endpoint model to verify if the device id is matching as a parent or child
+    if (is_success($status)){
+        my $device_name = $result->name;
+        my $endpoint = fingerbank::Model::Endpoint->new(name => $device_name, version => undef, score => undef);
+
+        for my $id (@oses) {
+            $logger->debug("The devices type ".$device_name." is authorized to be registered via the device-registration module");
```

Comment:
That logging statement is only true if the condition on the line below is true so that statement is false...

Wrap your return + this logging in the if block